packages/plugins/ganache/src/index.js

```diff
@@ -1,10 +1,41 @@
+const constants = require('embark-core/constants');
+
 class Ganache {
   constructor(embark) {
-    embark.events.request('blockchain:vm:register', () => {
+    embark.events.request("blockchain:node:register", constants.blockchain.clients.ganache, {
+      isStartedFn: (cb) => {
+        cb(null, true); // Always assume it's started, because it's just a provider (nothing to start)
+      },
+      launchFn: (cb) => {
+        cb(); // Will probably never be called, but if so, callback straight aways because there is nothing to start
+      },
+      stopFn: async (cb) => {
+        cb(); // Will probably never be called, but if so, callback straight aways because there is nothing to start
+      }
+    });
+
+    embark.events.request('blockchain:vm:register', constants.blockchain.clients.ganache, () => {
       const ganache = require('ganache-cli');
-      // Default to 8000000, which is the server default
-      // Somehow, the provider default is 6721975
-      return ganache.provider({gasLimit: '0x7A1200'});
+      const blockchainConfig = embark.config.blockchainConfig;
+
+      // Ensure the dir exists before initiating Ganache, because Ganache has a bug
+      // => https://github.com/trufflesuite/ganache-cli/issues/558
+      embark.fs.ensureDirSync(blockchainConfig.datadir);
+
+      const hasAccounts = blockchainConfig.accounts && blockchainConfig.accounts.length;
+
+      const isTest = embark.currentContext.includes('test');
+
+      return ganache.provider({
+        // Default to 8000000, which is the server default
+        // Somehow, the provider default is 6721975
+        gasLimit: blockchainConfig.targetGasLimit || '0x7A1200',
+        blockTime: blockchainConfig.simulatorBlocktime,
```

Comment:
should this be `blockchainConfig.clientConfig.simulatorBlocktime` ?

Reply:
Same difference. Everything in `clientConfig` gets merged into `blockchainConfig`. That was done back in the day, because it made it way simpler, since I didn't have to refactor every blockchain clients.